test/e2e/azure_test.go

```diff
@@ -450,6 +450,10 @@ var _ = Describe("Workload cluster creation", func() {
 				WaitForClusterIntervals:      e2eConfig.GetIntervals(specName, "wait-cluster"),
 				WaitForControlPlaneIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
 				WaitForMachinePools:          e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+				ControlPlaneWaiters: clusterctl.ControlPlaneWaiters{
+					WaitForControlPlaneInitialized:   InstallCloudProviderAzureHelmChart,
```

Comment:
Does this completely replace the helper to wait for certain conditions on the control planes or does is add the install cloud provider action _before_ waiting for the control plane to be initialized? ie. Are we still waiting for the Control Plane machines to be initialized + Ready in this test?

Reply:
I think what you may be implying is that we can use the default `WaitForControlPlaneMachinesReady` and simply "overwrite" the `WaitForControlPlaneMachinesReady` (which is the part where the missing providerID will short-circuit E2E progress). Is that right?

Reply:
Good find, will do

Reply:
locally tested out this approach, thanks for the suggestion, a big improvement!